Auth/Auth/Credentials/ICredentialManager.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System;
+using AeroGear.Mobile.Core.Storage;
+
+namespace AeroGear.Mobile.Auth.Credentials
+{
+    public interface ICredentialManager : IStorageManager<ICredential>
```

Comment:
I don't think `ICredentialManager` need to inherit  the `IStorageManager` interface, but rather it can use the `IStorageManager` interface if it needs too.  The reason is being that logically, a `CredentialManager` may or may not want to persist data on the disk. For example, during test, we don't need to persist data on device. 

So we should implement the `IStorageManager` for Android (using SharedPreferences) and iOS (UserPrefs perhaps?). This will also allow other modules to use them to persist data if they need to.

The the implementation of `ICredentialManager` can use `IStorageManager` to persist/delete credentials.

Does it make sense?

Reply:
@aidenkeating for now, let's just support string types. It's enough to meet our requirement.